### PR TITLE
[WIP] PR about chunking over (n, 1) arrays

### PR DESCRIPTION
**Pull request by kdmurray91**
See issue #295

### CODE REVIEW

**Comment by kdmurray91** [comment]
@FrancescAlted will need your (or another maintainer's) assistance implementing a fix

**Comment by kdmurray91** [comment]
@FrancescAlted the issue appears related to lines [201](https://github.com/Blosc/bcolz/blob/master/bcolz/chunked_eval.py#L201) and [212](https://github.com/Blosc/bcolz/blob/master/bcolz/chunked_eval.py#L212)  of  ``chunked_eval.py`` and [2210](https://github.com/Blosc/bcolz/blob/master/bcolz/carray_ext.pyx#L2210) of ``carray_ext.pyx``

It looks as though, when the chunked evaluation gets to the leftovers (2210 above), the 2-d `(n, 1)` array is appended to the temporary 1-d array (variable `out` in `_getrange`) resulting in the broadcast error.

File: bcolz/tests/test_carray.py [hunk 1]
```diff
@@ -1603,6 +1603,11 @@ def test12(self):
         self.assertTrue(type(cr) == np.ndarray)
         assert_array_equal(cr, nr, "eval does not work correctly")
 
+    def test13(self):
+        """Testing eval() with columnar [shape = (n, 1)] arrays"""
```
Comment (kdmurray91) [inline bcolz/tests/test_carray.py]: Also note that this test won't trigger the bug unless `self.N` is large enough that chunking is required.

**Comment by kdmurray91** [comment]
@FrancescAlted this PR has resolved my original issue in #295, but I'd like to discuss with you the fix. Do you use gitter/irc?

**Comment by FrancescAlted** [comment]
So the problems seems to be in  `_eval_blocks()`, which makes sense to me.  Good job!

I have just created a [Gitter](https://gitter.im/Blosc/bcolz) room in case you want more interactive discussion.

**Comment by kdmurray91** [comment]
@FrancescAlted I have raised some points in gitter

**Comment by FrancescAlted** [comment]
I did some quick benchmarks and I don't see a significant difference, so I merged that.  Thanks!

**Comment by kdmurray91** [comment]
Thanks very much @FrancescAlted. Is there any chance of you doing another 1.0.0-rc release containing this, and I'll test it on our production server.

**Comment by FrancescAlted** [comment]
Hey @kdmurray91 , have you had a chance to test RC2?  I'd like to release 1.0.0 final soon.

**Comment by kdmurray91** [comment]
AFAICT the original trigger for this bug is fixed, but I haven't had time to test it comprehensively. I'd say go with this version.

**Comment by kdmurray91** [comment]
@FrancescAlted hold the presses, I think there's still an issue. It works on our production box but not the cluster. I will evaluate and get back to you. Sorry about this!

**Comment by kdmurray91** [comment]
@FrancescAlted False alarm. This was a problem at our end.

Cheers,
Kevin

**Comment by FrancescAlted** [comment]
Excellent.  Thanks for testing.  Expect a release today or tomorrow.

**Pull request merged**